### DISCUSSION TITLE
Would it be feasible to modify the File callback to skip a file?

### DISCUSSION THREAD

**Discussion by ra-dave** [rikyoz/bit7z#142] [Q&A] [answered]
The return type from the File callback is void.  It would great if this could be changed to bool so that you could tell 7-Zip to skip processing that file.   Of course the 7-Zip API may not support that.  I'm not sure if it could somehow be handled by bit7z instead, in that case.

**Comment by rikyoz** [comment, marked as answer]
Currently, the only way to skip extracting a file is through the progress callback, which requires returning a boolean (`true` means continue, `false` stop).
Then, you can use both the progress and the file callbacks to achieve your desired behavior.

A file callback returning a boolean might be handy and would simplify your use case. However, it would also be a breaking change in the current API, which I didn't plan. I need to evaluate whether it's the case to implement it before releasing the stable v4; otherwise, it would require a new major release (I'm trying to follow the semantic versioning).

Anyway, thanks for the suggestion!

**Comment by ra-dave** [comment]
Skipping a file was more of a "nice to have" for me.  I can certainly work around not having that.
